packages/core/src/tools/facades/ToolsFactory.ts

```diff
@@ -0,0 +1,94 @@
+import { InternalInlineToolSettings, InternalTuneSettings } from './BaseToolFacade.js';
+import { InlineToolFacade } from './InlineToolFacade.js';
+import { BlockTuneFacade } from './BlockTuneFacade.js';
+import { BlockToolFacade } from './BlockToolFacade.js';
+// import type ApiModule from '../modules/api';
+import type {
+  ToolConstructable,
+  ToolSettings,
+  EditorConfig,
+  InlineToolConstructable,
+  BlockTuneConstructable
+} from '@editorjs/editorjs';
+
+type ToolConstructor = typeof InlineToolFacade | typeof BlockToolFacade | typeof BlockTuneFacade;
+
+export type UnifiedToolConfig = Record<string, Omit<ToolSettings, 'class'> & {
+  class: ToolConstructable;
+  isInternal?: boolean;
+}>;
+
+/**
+ * Factory to construct classes to work with tools
+ */
+export class ToolsFactory {
+  /**
+   * Tools configuration specified by user
+   */
+  private config: UnifiedToolConfig;
+
+  /**
+   * EditorJS API Module
+   */
+  private api: any;
```

Comment:
you can 
```ts
import { API } from '@editorjs/editorjs'
```

Reply:
Not in this PR, when API will be implemented in the core package